### PR TITLE
cli: Add "cilium bpf proxy list" command

### LINKED ISSUE #2277: Add `cilium bpf get proxymap`
The proxymap cannot be dumped from cilium commands currently. It'd be nice to have this ability for debugging purposes.

### PR DESCRIPTION
**Pull request by vadorovsky**
The command which dumps proxy BPF maps for debugging purposes.

Example (with Cilium, Envoy and Kubernetes):

```
$ kubectl create -f examples/kubernetes-grpc/cc-door-app.yaml
deployment "cc-door-mgr" created
service "cc-door-server" created
pod "terminal-87" created
$ kubectl create -f examples/kubernetes-grpc/cc-door-ingress-security.yaml
$ sudo cilium bpf proxy list
PROXY                             DESTINATION
10.0.118.183:39808 (6) => 10000   10.0.202.84:50051
10.0.118.183:39794 (6) => 10000   10.0.202.84:50051
10.0.118.183:39822 (6) => 10000   10.0.202.84:50051
```

Fixes #2277

Signed-off-by: Michal Rostecki <mrostecki@suse.com>

### CODE REVIEW

**Comment by aalemayhu** [comment]
test-me-please

**Review by aalemayhu — CHANGES_REQUESTED** [review]
Can you add an example of the output in the commit message? 10 lines would be good.

```
Proxy                Destination
--------------------------------
[f00d::a69:0:0:981b]:53208 (6) => 10000 f00d::a69:0:0:f236:80
[f00d::a69:0:0:981b]:58082 (6) => 10003 f00d::a69:0:0:f236:80
[f00d::a69:0:0:981b]:58056 (6) => 10000 f00d::a69:0:0:f236:80
[fc00::10ca:1]:40288 (6) => 10003 f00d::a69:0:0:dc06:80
[f00d::a69:0:0:981b]:58116 (6) => 10000 f00d::a69:0:0:f236:80
[f00d::a69:0:0:981b]:53756 (6) => 10000 f00d::a69:0:0:dc06:80
[f00d::a69:0:0:981b]:58730 (6) => 10000 f00d::a69:0:0:f236:80
[f00d::a69:0:0:981b]:58714 (6) => 10000 f00d::a69:0:0:f236:80
```

File: cilium/cmd/bpf_proxy_list.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		proxy.Dump6(dumpProxy6)
+
+		if len(dumpOutput) > 0 {
+			if err := OutputPrinter(proxyList); err != nil {
+				os.Exit(1)
+			}
+			return
+		}
+
+		title := fmt.Sprintf("%-20s %-s", "Proxy", "Destination")
```
Comment (joestringer) [inline cilium/cmd/bpf_proxy_list.go]: Some of the other CLI tools use tabwriter, and I think it's a good way to ensure that the columns being outputted line up. For example, with IPv6 it's easy for addresses to exceed 20 characters.

Would you mind updating this to do the same? Here's an example:

https://github.com/cilium/cilium/blob/b2f922369413d43ac592706f444af5dabf085ce9/cilium/cmd/bpf_policy_list.go#L82
https://github.com/cilium/cilium/blob/b2f922369413d43ac592706f444af5dabf085ce9/cilium/cmd/bpf_policy_list.go#L115
https://github.com/cilium/cilium/blob/b2f922369413d43ac592706f444af5dabf085ce9/cilium/cmd/bpf_policy_list.go#L134

**Review by joestringer — COMMENTED** [review]
Looks good to me, just one comment for the output of long lines - would be good to see the use of tabwriter. Thanks!

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by vadorovsky** [comment]
@joestringer done

**Comment by aalemayhu** [comment]
test-me-please

**Comment by vadorovsky** [comment]
@scanf I've just added an example in the commit message

File: cilium/cmd/bpf_proxy_list.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/proxy"
+)
```
Comment (aalemayhu) [inline cilium/cmd/bpf_proxy_list.go]: The order is supposed to be std, cilium and then 3rd party. So import cobra should be moved further down.

**Review by aalemayhu — APPROVED** [review]
One minor nitpick, can be fixed while rebasing. Rest LGTM, thanks.

**Comment by aalemayhu** [comment]
test-me-please

File: cilium/cmd/bpf_proxy_list.go (hunk 2 above)
Comment (vadorovsky) [reply cilium/cmd/bpf_proxy_list.go]: done

**Review by joestringer — APPROVED** [review]

**Review by manalibhutiyani — APPROVED** [review]
looks great! Thanks for doing this!

**Comment by joestringer** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**